packages/x-charts-pro/src/context/ZoomProvider/ZoomProvider.tsx

```diff
@@ -34,33 +21,109 @@ export function ZoomProvider({ children, xAxis, yAxis }: ZoomProviderProps) {
     [xAxis, yAxis],
   );
 
-  const [zoomData, setZoomData] = React.useState<ZoomData[]>(() =>
-    Object.values(options).map(({ axisId, minStart: start, maxEnd: end }) => ({
-      axisId,
-      start,
-      end,
-    })),
+  const [zoomData, setZoomData] = useControlled<ZoomData[]>({
+    controlled: zoom,
+    default: [],
```

Comment:
I think you need to define this array outside of the render function, otherwise `useControlled` will warn you that your default value changes across renders :+1:

Reply:
I didn't see any warning. Should I? 🤔 

I'm not sure it would be wise to create the array outside, as it could cause issues if someone edits it manually?

If we can't use the array there, it might be a better idea to allow `useControlled` to accept a function instead.

Reply:
My bad, we only run the effect when `JSON.stringify(defaultProp)` changes so we should not have the error on re-renders, feel free to ignore my comment